fastestimator/schedule/schedule.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+
+class Scheduler:
+    def get_current_value(self, epoch: int):
+        return None
+
+    def get_items(self):
+        raise NotImplementedError
+
+
+class RepeatScheduler(Scheduler):
+    def __init__(self, repeat_list):
+        assert isinstance(repeat_list, list), "must provide a list as input of RepeatSchedule"
+        self.repeat_list = repeat_list
+        self.cycle_length = len(repeat_list)
+        assert self.cycle_length > 1, "list length must be greater than 1"
+
+    def get_current_value(self, epoch: int):
+        return self.repeat_list[epoch % self.cycle_length]
+
+    def get_items(self):
+        return self.repeat_list
+
+
+class FunctionScheduler(Scheduler):
```

Comment:
this needs to implement get_items, or else the function needs to be removed from the base class

Reply:
yep I removed from base class